ui/ducks/app/app.ts

```diff
@@ -334,12 +334,12 @@ export default function reduceApp(
     case actionConstants.MODAL_CLOSE:
       return {
         ...appState,
-        modal: Object.assign(
-          appState.modal,
-          { open: false },
-          { modalState: { name: null, props: {} } },
-          { previousModalState: appState.modal.modalState },
-        ),
+        modal: {
+          ...appState.modal,
+          open: false,
+          modalState: { name: null, props: {} },
+          previousModalState: { ...appState.modal.modalState },
+        },
```

Comment:
Are you sure about this change? This looks like an important change to all close modals?

Reply:
Yep this is intentional. A legitimate bug.
https://github.com/MetaMask/metamask-extension/pull/24028/files/dce192fde23275dd43f7603cffb686841e4a2654#r1587392151